### DISCUSSION TITLE
Why are jsonc keys and values the same color?

### DISCUSSION THREAD

**Discussion by Avasam** [github-linguist/linguist#6197] [General]
It makes it harder to differentiate keys from values

```json
// Comment is invalid
{
  "foo": "bar",
  "foo2": true,
  "foo3": 3,
}
```
vs
```jsonc
// Comments are fine
{
  "foo": "bar",
  "foo2": true,
  "foo3": 3,
}
```

**Comment by lildude** [comment]
This is a syntax highlighting design question so really should be directed to the upstream maintainers.

From the [troubleshooting docs](https://github.com/github/linguist/blob/master/docs/troubleshooting.md#theres-a-problem-with-the-syntax-highlighting-of-a-file):

> ## There's a problem with the syntax highlighting of a file
> 
> Linguist detects the language of a file but the actual syntax-highlighting is powered by a set of language grammars which are included in this project as a set of submodules [as listed here](/vendor/README.md).
> 
> If you experience an issue with the syntax-highlighting on GitHub, **please report the issue to the upstream grammar repository, not here.** Grammars are updated every time we build the Linguist gem so upstream bug fixes are automatically incorporated as they are fixed.

**Reply by Avasam** [reply]
Ah, thanks for the link. There's the issue: JSON with comments is using javascript parsing. So it thinks it's just 2 strings (js string key) and makes no difference.

> JSON with Comments: [atom/language-javascript](https://github.com/atom/language-javascript)

```js
// js
{
  "foo": "bar",
  "foo2": true,
  "foo3": 3,
  valid:"js",
}
```

```jsonc
// jsonc
{
  "foo": "bar",
  "foo2": true,
  "foo3": 3,
  invalid:"json with comment",
}
```

```json5
// json5
{
  "foo": "bar",
  "foo2": true,
  "foo3": 3,
  valid:"json5",
}
```

```json
// json
{
  "foo": "bar",
  "foo2": true,
  "foo3": 3,
  invalid:"json",
}
```

Sidenote, that repo is about to be archived and abandoned  in ~2 weeks.

**Comment by Asomakn** [comment]
.

**Comment by DecimalTurn** [comment]
This issue will be fixed when https://github.com/github-linguist/linguist/pull/7255 is live on GitHub.

**Reply by DecimalTurn** [reply]
@Avasam, the fix is finally live on GitHub.

@RiskyMH, would you consider switching `bun.lock` to JSONC now that the syntax highlighting for [jsonc code](https://github.com/DecimalTurn/vscode-jsonc-syntax-highlighting/blob/main/sample.jsonc) is similar to JSON?

**Reply by RiskyMH** [reply]
If its the same look, I don't have any issue with it. But just note the only jsonc thing we do there is trailing comma which usually rendering is fine with (it's just parsing that doesn't like), so there isn't too much gain in switching.

On another note, thanks for doing this as it looks awesome for me to use in other places now!

**Reply by DecimalTurn** [reply]
> But just note the only jsonc thing we do there is trailing comma [...]

Ah, that explains why I couldn't find any bun.lock file on GitHub with comments 😅 ([search results](https://github.com/search?q=path%3A*bun.lock+%2F%5E%5Cs*%5C%2F%5C%2F%2F&type=code)).

In that case, yeah, there's not much gain from switching at the moment.
